internal/secrets/gcp_secret_manager.go

```diff
@@ -36,12 +51,24 @@ func NewGCPSecretManager(ctx context.Context) (SecretManager, error) {
 		return nil, fmt.Errorf("secretmanager.NewClient: %w", err)
 	}
 
-	return &GCPSecretManager{client}, nil
+	return &GCPSecretManager{
+		client: client,
+		cache:  make(map[string]*item),
+	}, nil
 }
 
 func (sm *GCPSecretManager) GetSecretValue(ctx context.Context, name string) (string, error) {
 	logger := logging.FromContext(ctx)
 
+	// Check cache.
+	sm.cacheMutex.RLock()
+	if i, ok := sm.cache[name]; ok && i.expiresAt <= time.Now().UnixNano() {
+		sm.cacheMutex.RUnlock()
+		logger.Debugf("found secret in cache: %v", name)
```

Comment:
can probably remove this debug statement

Reply:
the other part being the lookup and return

Reply:
Done